File: Documentation/5.4/Samples/python/ClientApi/Operations/Server/AddDatabaseNode.py
from ravendb.serverwide.operations.common import DatabasePutResult, ServerOperation

from examples_base import ExampleBase


class Foo:
    # region syntax
    class AddDatabaseNodeOperation(ServerOperation[DatabasePutResult]):
        def __init__(self, database_name: str, node_tag: str = None): ...

    # endregion


class AddDatabaseNode(ExampleBase):
    def setUp(self):
        super().setUp()

    def test_add_database_node(self):
        with self.embedded_server.get_document_store("AddDatabaseNode") as store:
            # region add_1
            # Create the AddDatabaseNodeOperation
            # Add a random node to 'Northwind' database-group
            add_database_node_op = AddDatabaseNodeOperation("Northwind")

            # Execute the operation by passing it to maintenance.server.send
            result = store.maintenance.server.send(add_database_node_op)

            # Can access the new topology
            number_of_replicas = len(result.topology.all_nodes)
            # endregion

            # region add_2
            # Create the AddDatabaseNodeOperation
            # Add node C to 'Northwind
            add_database_node_op = AddDatabaseNodeOperation("Northwind", "C")

            # Execute the operation by passing it to maintenance.server.send
            result = store.maintenance.server.send(add_database_node_op)
            # endregion

File: Documentation/5.4/Samples/php/ClientApi/Operations/Server/AddDatabaseNode.php
<?php

namespace RavenDB\Samples\ClientApi\Operations\Server;

use RavenDB\Documents\DocumentStore;
use RavenDB\ServerWide\Operations\AddDatabaseNodeOperation;
use RavenDB\ServerWide\Operations\DatabasePutResult;

class AddDatabaseNode
{
    public function samples(): void
    {
        $store = new DocumentStore();
        try {
            # region add_1
            // Create the AddDatabaseNodeOperation
            // Add a random node to 'Northwind' database-group
            $addDatabaseNodeOp = new AddDatabaseNodeOperation("Northwind");

            // Execute the operation by passing it to Maintenance.Server.Send
            /** @var DatabasePutResult $result */
            $result = $store->maintenance()->server()->send($addDatabaseNodeOp);

            // Can access the new topology
            $numberOfReplicas = count($result->getTopology()->getMembers());
            # endregion
        } finally {
            $store->close();
        }

        $store = new DocumentStore();
        try {
            # region add_2
            // Create the AddDatabaseNodeOperation
            // Add node C to 'Northwind' database-group
            $addDatabaseNodeOp = new AddDatabaseNodeOperation("Northwind", "C");

            // Execute the operation by passing it to Maintenance.Server.Send
            /** @var DatabasePutResult $result */
            $result = $store->maintenance()->server()->send($addDatabaseNodeOp);
            # endregion
        } finally {
            $store->close();
        }
    }
}

/*
interface IFoo
{
    # region syntax
    AddDatabaseNodeOperation(?string $databaseName, ?string $nodeTag = null)
    # endregion
}
*/
